pkg/upload/unity-android.go

```diff
@@ -0,0 +1,162 @@
+package upload
+
+import (
+	"fmt"
+	"github.com/bugsnag/bugsnag-cli/pkg/android"
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/server"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type UnityAndroidOptions struct {
+	AabPath       string            `help:"Path to Android AAB file"`
+	ApplicationId string            `help:"Module application identifier"`
+	Arch          string            `help:"The architecture of the shared object that the symbols are for (e.g. x86, armeabi-v7a)."`
+	Path          utils.UploadPaths `arg:"" name:"path" help:"(required) Path to directory or file to upload" type:"path"`
+	ProjectRoot   string            `help:"path to remove from the beginning of the filenames in the mapping file" type:"path"`
+	VersionCode   string            `help:"The version code for the application (Android only)."`
+	VersionName   string            `help:"The version name of the application."`
+	BuildUuid     string            `help:"The version name of the application."`
+}
+
+func ProcessUnityAndroid(apiKey string, aabPath string, applicationId string, versionCode string, buildUuid string, arch string, versionName string, projectRoot string, paths []string, endpoint string, failOnUploadError bool, retries int, timeout int, overwrite bool, dryRun bool) error {
+	var err error
+	var zipPath string
+	var archList []string
+	var symbolFileList []string
+	var manifestData map[string]string
+	var aabManifestPath string
+
+	for _, path := range paths {
+		if utils.IsDir(path) {
+			zipPath, err = utils.FindFileWithSuffix(path, ".symbols.zip")
+
+			if err != nil {
+				return err
+			}
+
+			if aabPath == "" {
+				aabPath, err = utils.FindFileWithSuffix(path, ".aab")
+
+				if err != nil {
+					return err
+				}
+			}
+		} else if strings.HasSuffix(path, ".symbols.zip") {
+			zipPath = path
+
+			if aabPath == "" {
+				buildDirectory := filepath.Dir(path)
+
+				aabPath, err = utils.FindFileWithSuffix(buildDirectory, ".aab")
+
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			return fmt.Errorf("unsupported files. Please specify the unity `symbols.zip` file.")
+		}
+	}
+
+	log.Info("Using " + aabPath + " as the Unity Android AAB file")
+
+	log.Info("Using " + zipPath + " as the Unity Android symbols zip file")
+
+	aabDir, err := utils.ExtractFile(aabPath, "aab")
+
+	if err != nil {
+		return err
+	}
+
+	defer os.RemoveAll(aabDir)
+
+	aabManifestPathExpected := filepath.Join(aabDir, "base", "manifest", "AndroidManifest.xml")
+	if utils.FileExists(aabManifestPathExpected) {
+		aabManifestPath = aabManifestPathExpected
+	} else {
+		log.Warn("AndroidManifest.xml not found in AAB file")
+	}
+
+	if aabManifestPath != "" && (applicationId == "" || buildUuid == "" || versionCode == "" || versionName == "") {
+		log.Info("Reading data from AndroidManifest.xml")
```

Comment:
This is leaking the knowledge of the AAB structure outside the AAB commands - suggest moving L77-87 inside the new AAB manifest reading function as it's common to both.